spacelift/data_worker_pools.go

```diff
@@ -0,0 +1,94 @@
+package spacelift
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/pkg/errors"
+
+	"github.com/spacelift-io/terraform-provider-spacelift/spacelift/internal"
+	"github.com/spacelift-io/terraform-provider-spacelift/spacelift/internal/structs"
+)
+
+func dataWorkerPools() *schema.Resource {
+	return &schema.Resource{
+		Read: dataWorkerPoolsRead,
+		Schema: map[string]*schema.Schema{
+			"worker_pools": &schema.Schema{
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"worker_pool_id": {
+							Type:        schema.TypeString,
+							Description: "ID of the worker pool",
+							Computed:    true,
+						},
+						"config": {
+							Type:        schema.TypeString,
+							Description: "credentials necessary to connect WorkerPool's workers to the control plane",
+							Sensitive:   true,
+							Computed:    true,
+						},
+						"name": {
+							Type:        schema.TypeString,
+							Description: "name of the worker pool",
+							Computed:    true,
+						},
+						"description": {
+							Type:        schema.TypeString,
+							Description: "description of the worker pool",
+							Computed:    true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataWorkerPoolsRead(d *schema.ResourceData, meta interface{}) error {
+	var query struct {
+		WorkerPools []*structs.WorkerPool `graphql:"workerPools()"`
+	}
+	variables := map[string]interface{}{}
+
+	if err := meta.(*internal.Client).Query(&query, variables); err != nil {
+		return errors.Wrap(err, "could not query for worker pools")
+	}
+
+	workerPools := query.WorkerPools
+	if workerPools == nil {
+		return nil
+	}
+
+	wps := flattenDataWorkerPoolsList(workerPools)
+	if err := d.Set("worker_pools", wps); err != nil {
```

Comment:
Do we want to set `worker_pools` to nil here as well? @cube2222 @marcinwyszynski

Reply:
Why so?

Reply:
You mean if the error is non-nil?

Reply:
@michieldewilde

Reply:
It's just set ID to "", which will remove the entire resource.

Reply:
My idea is that there are two scenarios:
- no worker_pools found => set worker_pools resource to nil and keep the resource ID
- generic error => set resource ID to `""` and return error